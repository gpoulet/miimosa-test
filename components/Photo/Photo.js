import CardMedia from '@material-ui/core/CardMedia'
import { string } from 'prop-types'
import Link from 'next/link'

function media(id, label) {
  return (
    <CardMedia
      title={label}
      style={{ height: 0, paddingTop: '40%' }}
      image={"/images/" + id + ".png"}/>
  )
}

function Photo({ id, label, href }) {
  if (href) {
    return (
      <Link href={href}>
        <a>
          {media(id, label)}
        </a>
      </Link>
    )
  } else {
    return (
      <>{media(id, label)}</>
    )
  }
}

Photo.propTypes = {
  id: string.isRequired,
  label: string.isRequired,
  href: string,
}

export default Photo;