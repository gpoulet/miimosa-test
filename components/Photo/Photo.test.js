import React from 'react';
import renderer from 'react-test-renderer';
import Photo from './Photo';

it('Photo renders correctly', () => {
  const tree = renderer
    .create(<Photo id="test" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});