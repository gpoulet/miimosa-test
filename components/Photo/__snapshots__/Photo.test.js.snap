// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo renders correctly 1`] = `
<div
  className="MuiCardMedia-root"
  style={
    Object {
      "backgroundImage": "url(\\"/images/test.png\\")",
      "height": 0,
      "paddingTop": "40%",
    }
  }
/>
`;
